searxng_extra/update/update_locales.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Update locale names in :origin:`searx/data/locales.json` used by
+:ref:`searx.locales`
+
+- :py:obj:`searx.locales.RTL_LOCALES`
+- :py:obj:`searx.locales.LOCALE_NAMES`
+"""
+from __future__ import annotations
+
+from typing import Set
+import json
+from pathlib import Path
+import os
+
+import babel
+import babel.languages
+import babel.core
+
+from searx import searx_dir
+from searx.locales import (
+    ADDITIONAL_TRANSLATIONS,
+    LOCALE_BEST_MATCH,
+    get_translation_locales,
+)
+
+LOCALE_DATA_FILE = Path(searx_dir) / 'data' / 'locales.json'
```

Comment:
The `locales.json` is confusing: 
* the same data are displayed to the user as `Interface language`.
* there is already a generated file named `sxng_locales.py`.

Reply:
> The `locales.json` name is confusing:
> * the same data are displayed to the user as `Interface language`.

What do you had in mind / I have only this part of code in mind

https://github.com/searxng/searxng/blob/c197c0e35e6eaa823d6ba8606df4e5a5c598a07b/searx/webapp.py#L1004

and I would prefer to change the name `locales` in the templates to `locale_names` or better `LOCALE_NAMES`. / but templates are not in the scope of this PR.

Reply:
> I would prefer to change the name locales in the templates to locale_names

Including in the UI? That's not locales, that's language for the UI, that's why SearXNG needs the labels : to display a list of languages for the UI.

Reply:
>  That's not locales, that's language for the UI

Sorry, my fail .. you are right; "languages" or similar would be a better name / back to your initial comment:

> The `locales.json` name is confusing: 

IMO its a name that fits too its usage .. it contains values for names that are defined in the `searx.locales` module.

https://github.com/searxng/searxng/blob/ed66ed758dbd1c926296aa227da8a82fff9166e1/searx/locales.py#L54-L63

I had thought about naming the file `locales_data.json` ... but found that a bit superfluous, because the file is already in a folder `data/`